webapp/src/ts/services/enketo.service.ts

```diff
@@ -288,7 +288,7 @@ export class EnketoService {
       });
   }
 
-  private renderFromXmls(doc, wrapper, instanceData) {
+  private renderFromXmls({ doc, wrapper, instanceData, titleKey }) {
```

Comment:
I like this idea, it makes the function parameters definition shorter.  
One of the drawbacks is that it makes harder to know which parameters we can pass to the function :sweat_smile: 

Is there any way to keep it short, easy to read and easy to discover which parameters you can pass?

Reply:
We moved to typescript few months ago, so this is not wide spread in the code yet, but one thing I like to do is defining [an interface ](https://github.com/medic/cht-core/blob/0f69b0c76a108892fb70b38ae43d5c40f80da278/webapp/src/ts/services/uhc-stats.service.ts#L106) or a type (specially for public functions) The IDE provides accurate suggestions when we add these.

Reply:
We could probably define a type like this
```ts
type Await<T> = T extends Promise<infer U> ? U : T;
type AsyncReturnType<T extends (...args: any) => any> = Await<ReturnType<T>>;

type ContextXMLForm = {
  doc: AsyncReturnType<typeof EnketoService['prototype']['transformXml']>;
  wrapper: JQuery;
  instanceData: ReturnType<typeof ContactsEditComponent['prototype']['getFormInstanceData']>;
  titleKey: string;
}

private renderFromXmls(contextXMLForm: ContextXMLForm) {}
```

Reply:
is `ContextXMLForm` a type or an interface?

`doc` is kind of tricky to define, the `AsyncReturnType` isn't very descriptive on what it actually is... I wonder if we can do something like: 

dbDoc {
   _id: string;
   rev: string;
  [name: string]: any;  /// like to say that after _id and rev, any other key is fine with any value....
}

interface formDoc extends dbDoc {
   attachments: etc.....
}

regarding `instanceData` could be anything right? 🤔 as long as it's an object. 

For us, it can get complex when typing because of the code dependency..  if it gets too complex, no worries about it :)

Reply:
I just pushed a commit to type this and make it more readable while limiting the typing complexity haha, let me know what you think about it

Reply:
Nice!